### PR TITLE
Fixes issues with filter tree, moves query basic to use filterTree directly

### PR DESCRIPTION
**Pull request by andrewkfiedler**
See commits for breakdown.  Big part of this was enforcing classes, which involved making sure typescript didn't allow using the spread operator (which results in a plain object) without feeding that into a new instance of the class.

Also optimized some bits for performance since depending on the filter tree directly caused more frequent updates to it.  Notably there is a change in the home component, and we now memo the query settings.

Testing:
Go wild testing everything on the basic page, then test going to the advanced and ensuring it translates appropriately.

In advanced, test out grouping, negation, etc.

### CODE REVIEW

**Comment by vinamartin** [comment]
🎉  Hero successful- extra kudos on fixing a lot of the crashing issues I've been seeing with query basic time.
One minor issue I saw that I also saw on my PR and have been trying to fix is that you can't "Clear All" these time properties using the "X" button **and** you also can't remove the last one. It's kind of minor but it's a pain if you only want one of the properties and have to manually click on all of the others to clear them instead of just clearing and typing what you want.
Screen Shot 2020-10-29 at 07 17 38

I think what's happening here is that we're trying to figure out what kind of filter it is on update and it doesn't work with having undefined property values. That's what I was trying to fix here: https://github.com/codice/ddf-ui/pull/437. 
I think I'm going to close out that PR in favor of writing up a patch against this one.

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/react-component/location/keyword.spec.js [hunk 1]
```diff
@@ -33,51 +33,51 @@ const { shallow } = Enzyme
...
-    const fetch = async () => {
-      throw new Error('unavailable')
-    }
-    const wrapper = shallow(<Keyword fetch={fetch} onError={onError} />)
-    wrapper.find('AutoComplete').prop('onChange')({
-      id: '0',
-      name: 'test',
-    })
-  })
+  // it('should fetch features on select', (done) => {
```
Comment (jordanwilking) [inline ui-frontend/packages/catalog-ui-search/src/main/webapp/react-component/location/keyword.spec.js:41]: ✏️ Should we just delete all of these tests if we don't intend to replace them now?

**Review by jordanwilking — APPROVED** [review]

**Review by vinamartin — APPROVED** [review]

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/react-component/location/keyword.spec.js (hunk 1 above)
Comment (andrewkfiedler) [reply ui-frontend/packages/catalog-ui-search/src/main/webapp/react-component/location/keyword.spec.js:41]: I'm not sure to be honest, I left them commented out as a guide for what we should try and fix.  They've been not running for so long that I'm not entirely sure how relevant they are, but they probably have good ideas of what we should be testing in the components (the components have just changed too much). 

I originally had wanted to add some tests to ensure the filter builder didn't regress, but ran into issues just trying to render it.  Didn't have time to chase them down, but thought I'd include the change to make it possible to turn tests back on.

Comment (jordanwilking) [reply ui-frontend/packages/catalog-ui-search/src/main/webapp/react-component/location/keyword.spec.js:41]: We could always check an earlier commit to reference these, but it's not a big deal me 🤷. I'm sure we'll revisit these as we increase our testing efforts soon.

**Comment by andrewkfiedler** [comment]
@vinamartin nice catch, I think I can do something similar to what I did for types (I had to make a fake '*' 😭 )

**Comment by andrewkfiedler** [comment]
@vinamartin could you rehero this?  I pushed up a fix for the issue you saw, and added the ability to press enter to run a search (if within a string field on advanced search).

**Comment by vinamartin** [comment]
Manually verified ✨ 
Looks good to me, merging

**Pull request merged**